pom.xml

```diff
@@ -82,6 +82,24 @@
          </configuration>
        </plugin>
      </plugins>
+    <pluginManagement>
+      <plugins>
+      <plugin>
+        <groupId>org.jenkins-ci.tools</groupId>
+        <artifactId>maven-hpi-plugin</artifactId>
+        <extensions>true</extensions>
+        <configuration>
+          <archive>
+            <index>true</index>
+            <manifestEntries>
+              <Hudson-Version>1.609.18.1</Hudson-Version>
+              <Jenkins-Version>1.609.18.1</Jenkins-Version>
```

Comment:
Of course one problem with a reflection-based approach is that `hpi:run` and any automated in-plugin tests will be using an old core that does not reflect the behavior of current users. Acceptance tests would be the only things catching “facepalm” regressions.